source/auth/tests/connection-string.json

```diff
@@ -376,6 +376,63 @@
       "description": "should throw an exception if no username/password provided (userinfo implies default mechanism)",
       "uri": "mongodb://:@localhost.com/",
       "valid": false
+    },
+    {
+      "description": "should recognise the mechanism (MONGODB-AWS)",
+      "uri": "mongodb://localhost/?authMechanism=MONGODB-AWS",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
```

Comment:
All the existing tests use a `credential` field that encompasses all this information.  Why is this different?

Reply:
Fixed